image.md

```diff
@@ -0,0 +1,308 @@
+While the runtime only sees a single directory as the root filesystem, it might actually be stored as a layering of filesytems under the covers.
+When representing this layering within an image, each "layer" will be independently referenced.
+
+## Image Contents
+
+A compliant image contains the following pieces of information:
+
+1. A `manifest.json` file describing the contents of the image.
+This file is REQUIRED to be present in the image.
+This file extends the [`config.json`](config.md) file with some additional properties:
+  * `name`: This REQUIRED property is a URL-like string to aide in discovery of the image.
+  For example, `example.com/myApp`.
+  * `whiteout`: This OPTIONAL property is a JSON array of absolute file paths that are to be considered to be deleted from dependent layers, even though they are included within those layers root filesystems.
+  It is an implementation detail to decide how this whiteout list is processed, but an implementation MUST ensure that any container that is created from this image will behave as if those files were not in the root filesystem at all.
+  * `layers`: This OPTIONAL property contains references to the layers that make up the root filesystem of the image.
+  This property MUST be a JSON encoded array where each entry consists of a REQUIRED `name` and OPTIONAL `ID` properties.
+  The array MUST be in the order in which the images are to be "layed down" into the filesystem bundle that is generated from this image.
+  In other words, the first image in the array is written to the filesystem first, and subsequent images are overlayed on top - potentially overlaying an earlier image's files.
+  If a referenced layer is embedded within this image then an `ID` property MUST be included.
+  If a referenced layer is not embedded within this image then an `ID` property is OPTIONAL.
+  * `root/path`: ThiS REQUIRED property, defined within [`config.json`](config.md), is the name of the directory into which all layers MUST appear.
+  While the `manifest.json` file extends the `config.json` file, a compliant OCI image is NOT REQUIRED to have all of the required fields from `config.json` within its `manifest.json` file if that image is only used as a layer (dependent image) within other images.
+  This is due to the fact that a layer's `manifest.json` properties are not used to produce a container's `config.json` file and therefore not all of them are used.
+  However, if an image is used to create a container then all of the "required" fields of the `config.json` MUST be present in the `manifest.json`.
+
+1. Layers.
+Each embedded layer MUST have a corresponding image (tar) file, at the root of the image.
+Each layer's tar file MUST be named with its ID followed by the appropriate file-extension based on how the tar file is written (e.g. `.tar` for uncompressed).
```

Comment:
I wasn't focusing on the extension for any purpose, other than to say it needs to have one **after** the ID part of the name.

Not sure what your "why?" was referring to, but if its the ID part of the filename then I did that because we needed a way to go from the "layers" to the embedded tar file in a consistent fashion.  We can't use the "name" of the image because it might have characters that are not permitted on all filesystems - like a "/" - so having a tar file with that name might not be possible.  Using, and requiring an ID, also forces us to ensure there's a crypto key for each embedded image.